pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -185,6 +185,40 @@ func (configgen *ConfigGeneratorImpl) buildGatewayListeners(builder *ListenerBui
 	return builder
 }
 
+func buildNameToServiceMapForHTTPRoutes(node *model.Proxy, push *model.PushContext,
+	virtualService config.Config) map[host.Name]*model.Service {
+
+	vs := virtualService.Spec.(*networking.VirtualService)
+	nameToServiceMap := map[host.Name]*model.Service{}
+
+	addService := func(hostname host.Name) {
+		if _, exist := nameToServiceMap[hostname]; exist {
+			return
+		}
+		// First, we obtain the service which has the same namespace as virtualService
+		if s, exist := push.ServiceIndex.HostnameAndNamespace[hostname][virtualService.Namespace]; exist {
```

Comment:
There is a rare case here, if the service within the virtual service is not exported to the proxy should fall back. Otherwise, lgtm

Reply:
Make sense. I got it. Thanks